client/collection.go

```diff
@@ -0,0 +1,160 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/tigrisdata/tigris-client-go/driver"
+	"github.com/tigrisdata/tigris-client-go/filter"
+	"github.com/tigrisdata/tigris-client-go/projection"
+	"github.com/tigrisdata/tigris-client-go/schema"
+	"github.com/tigrisdata/tigris-client-go/update"
+)
+
+var (
+	errNotFound = fmt.Errorf("document not found")
+)
+
+type Collection[T schema.Model] struct {
+	name   string
+	driver driver.Driver
+	db     *Database
+	schema *schema.Schema
+	model  interface{}
+	crud   driver.Database
+}
+
+// Drop drops the collection
+func (c *Collection[T]) Drop(ctx context.Context) error {
+	return c.crud.DropCollection(ctx, c.name)
+}
+
+// Insert inserts documents into the collection.
+// Returns error if the documents with the same primary key exists already
+func (c *Collection[T]) Insert(ctx context.Context, docs ...*T) error {
+	var err error
+
+	bdocs := make([]driver.Document, len(docs))
+	for k, v := range docs {
+		bdocs[k], err = json.Marshal(v)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = c.crud.Insert(ctx, c.name, bdocs)
+
+	return err
+}
+
+// InsertOrReplace insert new documents and in the case of duplicate key
+// replaces existing documents with the new document
+func (c *Collection[T]) InsertOrReplace(ctx context.Context, docs ...*T) error {
+	var err error
+
+	bdocs := make([]driver.Document, len(docs))
+	for k, v := range docs {
+		bdocs[k], err = json.Marshal(v)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = c.crud.Replace(ctx, c.name, bdocs)
+
+	return err
+}
+
+// Update partially updates documents based on the provided filter
+// and provided document mutation
+func (c *Collection[T]) Update(ctx context.Context, filter filter.Filter, update *update.Update) error {
+	f, err := filter.Build()
+	if err != nil {
+		return err
+	}
+	u, err := update.Build()
+	if err != nil {
+		return err
+	}
+	_, err = c.crud.Update(ctx, c.name, f, u)
```

Comment:
response?

Reply:
I was thinking to fill schema.Model out of this response later. If it's embedded into user struct.